pkgs/by-name/va/vatprism/package.nix

```diff
@@ -0,0 +1,109 @@
+# Made in accordance to https://fzakaria.com/2020/07/20/packaging-a-maven-application-with-nix.html
+{ lib
+, stdenv
+, jdk
+, maven
+, makeWrapper
+, fetchFromGitHub
+, libGL
+, libxkbcommon
+, wayland
+, fontconfig
+, xorg
+}: let
+  pname = "vatprism";
+  version = "0.3.5";
+  src = fetchFromGitHub {
+    owner = "marvk";
+    repo = "vatprism";
+    rev = "v${version}";
+    hash = "sha256-ofEwHUCm79roHe2bawmKFw2QHhIonnlkFG5nhE6uN+g=";
+  };
+  dependencies = stdenv.mkDerivation {
+    name = "${pname}-dependencies1";
+    nativeBuildInputs = [ jdk maven ];
+    # https://github.com/marvk/vatprism/pull/141
+    patches = [ ./0001-Fix-build-on-JDK-21.patch ];
+    inherit src;
+    buildPhase = ''
+      runHook preBuild
+      while mvn package -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000; [ $? = 1 ]; do
+        echo "timeout, restart maven to continue downloading"
+      done
+      runHook postBuild
+    '';
+    # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+    installPhase = ''
+      runHook preInstall
+      find $out/.m2 -type f -regex '.+\\(\\.lastUpdated\\|resolver-status\\.properties\\|_remote\\.repositories\\)' -delete
+      runHook postInstall
+    '';
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "zHPSFz8GggFWxrIkJF2DmB1OI/y3UX1AOFBCaVvDK3Y=";
+  };
+  libPath = lib.makeLibraryPath [
+    libGL
+    libxkbcommon
+    wayland
+    xorg.libX11
+    xorg.libXcursor
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXxf86vm
+    xorg.libXtst
+    fontconfig
+  ];
+in
+  stdenv.mkDerivation rec {
```

Comment:
why do we not use `buildMavenPackage` here?
https://nixos.org/manual/nixpkgs/unstable/#maven

Reply:
Didn't know that existed! Didn't look enough I suppose.